src/route/dataset.ts

```diff
@@ -523,6 +760,75 @@ router.get(
     }
 );
 
+// POST /:dataset_id/dimension/by-id/:dimension_id/lookup
+// Attaches a lookup table to do a dimension and validates
+// the lookup table.
+router.post(
+    '/:dataset_id/dimension/by-id/:dimension_id/lookup',
+    upload.single('csv'),
+    loadDataset(),
+    async (req: Request, res: Response, next: NextFunction) => {
+        if (!req.file) {
+            next(new BadRequestException('errors.upload.no_csv'));
+            return;
+        }
+        const dataset = res.locals.dataset;
+        if (!dataset) {
+            next(new NotFoundException('errors.dataset_id_invalid'));
+            return;
+        }
```

Comment:
I've taken this out... Its because I largely reused the file upload code and extended it to handle the extra dimension stuff after... Again a refactor here could save a few lines of code as there's now 2 routes which share a good chunk of code.

I am also thinking when we refactor we should apply the same pattern you've used on the frontend where the route only lists the routes and loads the middleware before passing on to a function in another file.  The dataset route is getting very long now.